File: app/src/main/java/com/novel/read/widget/HeadLayout.java
package com.novel.read.widget;

import android.content.Context;
import android.content.res.TypedArray;
import android.graphics.drawable.Drawable;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.annotation.Nullable;
import androidx.core.content.ContextCompat;

import com.novel.read.R;

public class HeadLayout extends LinearLayout {

    private TextView mTvHead;
    private TextView mTvMore;
    public HeadLayout(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
        LayoutInflater.from(context).inflate(R.layout.widget_head, this, true);
        initViews();
        initAttr(attrs);
    }


    private void initViews() {
        mTvHead=findViewById(R.id.tv_head);
        mTvMore = findViewById(R.id.tv_more);
    }
    private void initAttr(AttributeSet attrs) {
        TypedArray typedArray = getContext().obtainStyledAttributes(attrs, R.styleable.HeadLayout);
        boolean showHead = typedArray.getBoolean(R.styleable.HeadLayout_showHead, true);
        int headText=typedArray.getResourceId(R.styleable.HeadLayout_headText,R.string.empty_info);
        int headImg= typedArray.getResourceId(R.styleable.HeadLayout_headImg, R.mipmap.ic_launcher);
        boolean showRight = typedArray.getBoolean(R.styleable.HeadLayout_showRightText, true);
        int rightText=typedArray.getResourceId(R.styleable.HeadLayout_rightText,R.string.empty_info);
        int rightImg=typedArray.getResourceId(R.styleable.HeadLayout_rightImg,R.mipmap.ic_launcher);
        if (showHead){
            mTvHead.setText(headText);
            Drawable drawable = ContextCompat.getDrawable(getContext(), headImg);
            drawable.setBounds(0, 0, drawable.getMinimumWidth(), drawable.getMinimumHeight());
            mTvHead.setCompoundDrawables(drawable, null, null, null);
        }else {
            mTvHead.setVisibility(GONE);
        }
        if (showRight){
            mTvMore.setText(rightText);
            if (rightImg!=R.mipmap.ic_launcher){
                Drawable drawable = ContextCompat.getDrawable(getContext(), rightImg);
                drawable.setBounds(0, 0, drawable.getMinimumWidth(), drawable.getMinimumHeight());
                mTvHead.setCompoundDrawables(null, null, drawable, null);
            }
        }else {
            mTvMore.setVisibility(GONE);
        }

        typedArray.recycle();
    }

    //设置点击右边文字的监听
    public void setRightTextClickListener(OnClickListener listener) {
        mTvMore.setOnClickListener(listener);
    }


}

File: app/src/main/java/com/novel/read/widget/HeadLayout.kt
package com.novel.read.widget

import android.content.Context
import android.util.AttributeSet
import android.view.LayoutInflater
import android.view.View
import android.widget.LinearLayout
import androidx.core.content.ContextCompat
import com.novel.read.R
import kotlinx.android.synthetic.main.widget_head.view.*

class HeadLayout(context: Context, attrs: AttributeSet?) : LinearLayout(context, attrs) {

    init {
        LayoutInflater.from(context).inflate(R.layout.widget_head, this, true)
        initAttr(attrs)
    }

    private fun initAttr(attrs: AttributeSet?) {
        val typedArray = context.obtainStyledAttributes(attrs, R.styleable.HeadLayout)
        val showHead = typedArray.getBoolean(R.styleable.HeadLayout_showHead, true)
        val headText = typedArray.getResourceId(R.styleable.HeadLayout_headText, R.string.empty_info)
        val headImg = typedArray.getResourceId(R.styleable.HeadLayout_headImg, R.mipmap.ic_launcher)
        val showRight = typedArray.getBoolean(R.styleable.HeadLayout_showRightText, true)
        val rightText = typedArray.getResourceId(R.styleable.HeadLayout_rightText, R.string.empty_info)
        val rightImg = typedArray.getResourceId(R.styleable.HeadLayout_rightImg, R.mipmap.ic_launcher)
        if (showHead) {
            tv_head.setText(headText)
            val drawable = ContextCompat.getDrawable(context, headImg)
            drawable!!.setBounds(0, 0, drawable.minimumWidth, drawable.minimumHeight)
            tv_head.setCompoundDrawables(drawable, null, null, null)
        } else {
            tv_head.visibility = View.GONE
        }
        if (showRight) {
            tv_more.setText(rightText)
            if (rightImg != R.mipmap.ic_launcher) {
                val drawable = ContextCompat.getDrawable(context, rightImg)
                drawable!!.setBounds(0, 0, drawable.minimumWidth, drawable.minimumHeight)
                tv_head.setCompoundDrawables(null, null, drawable, null)
            }
        } else {
            tv_more.visibility = View.GONE
        }

        typedArray.recycle()
    }

    //设置点击右边文字的监听
    fun setRightTextClickListener(listener: View.OnClickListener) {
        tv_more.setOnClickListener(listener)
    }


}
